### PR TITLE
chore: update sonic version & enhance build logger

### LINKED ISSUE #750: go1.23 build fail
### Describe your question

error message
```
[ERROR] run: Command 'GOOS=linux GOARCH=amd64 go build -mod=mod -o /root/open_gemini/src/openGemini/build/ts-cli -ldflags="-X github.com/openGemini/openGemini/app.Version=1.2.0 -X github.com/openGemini/openGemini/app.GitBranch=main -X github.com/openGemini/openGemini/app.GitCommit=8797d1e89fa48271f6e6428de97e2ee83d57ea99" ./app/ts-cli' failed with error: b'# github.com/openGemini/openGemini/app/ts-cli\nlink: github.com/bytedance/sonic/ast: invalid reference to encoding/json.safeSet\n'
```

go version go1.23.1 linux/amd64
related to [sonic ](https://github.com/bytedance/sonic/issues/660). Upgrade to sonic 1.12.2 may fix the problem. 

Besides, should we check the `content` type in [build.py](https://github.com/openGemini/openGemini/blob/main/build.py#L407) since `content` may be bytes.

### PR DESCRIPTION
**Pull request by xwyzsn**
### What problem does this PR solve?

fix #750 

### What is changed and how it works?
update the sonic version for go1.23 and enhance the build logging

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B
- [ ] Test cases to be added
- [X] No code

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,5 @@
 module github.com/openGemini/openGemini
 
-go 1.20
```
Comment (xuthus5) [inline go.mod:3]: The semantics that specify the minimum version of Go required to compile the packages in this module are invalidated.

**Comment by xuthus5** [comment]
We need to discuss the handling strategy for `sonic`, or upgrade the minimum version of Go required. cc @xiangyu5632

**Comment by xiangyu5632** [comment]
> We need to discuss the handling strategy for `sonic`, or upgrade the minimum version of Go required. cc @xiangyu5632

We will not upgrade the Go version to v1.23 for system stability unless some necessary third-party components are imported and a later Go version is required.

**Comment by xuthus5** [comment]
Nothing happened, did you miss the code commit.

File: go.mod (hunk 1 above)
Comment (xwyzsn) [reply go.mod:3]: Should we keep this? 
(https://github.com/openGemini/openGemini/issues/750#issuecomment-2362993092)

Comment (xuthus5) [reply go.mod:3]: Yes.

**Comment by xwyzsn** [comment]
Hi, please take some time to review it.

**Review by xuthus5 — APPROVED** [review]

**Review by xiangyu5632 — APPROVED** [review]

**Comment by xiangyu5632** [comment]
tks @xwyzsn

**Pull request merged**